From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: jonakls <asmot54@hotmail.com>
Date: Thu, 8 Jun 2023 20:39:21 -0500
Subject: [PATCH] Remove toString method


diff --git a/src/main/java/org/bukkit/command/defaults/PluginsCommand.java b/src/main/java/org/bukkit/command/defaults/PluginsCommand.java
index 6f2eef45b72dc68668aa214cc21a08997b477eb5..052c7ab80b4e3290bbed6d925ad1ed2d89e61e19 100644
--- a/src/main/java/org/bukkit/command/defaults/PluginsCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/PluginsCommand.java
@@ -144,8 +144,7 @@ public class PluginsCommand extends BukkitCommand {
             pluginList.append(plugin.isEnabled() ? ChatColor.GREEN : ChatColor.RED);
             pluginList.append(plugin.getDescription().getName());
         }
-
-        return "(" + plugins.length + "): " + pluginList.toString();
+        return "(" + plugins.length + "): " + pluginList; // StelarPlex - remove unnecessary toString()
     }
 
     // Spigot Start
